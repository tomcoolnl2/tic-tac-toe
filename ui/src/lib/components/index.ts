export { AppLogo } from './logo/logo';
export { Avatar } from './avatar/avatar';
export { Button } from './button/button';
export { Cell } from './cell/cell';
export { ChooseDifficulty } from './choose-difficulty/choose-difficulty';
export { GameOver } from './game-over/game-over';
export { Input } from './input/input';
export { Modal } from './modal/modal';
export { ReloadButton } from './reload-button/reload-button';
export { ScoreBoardItem } from './score-board-item/score-board-item';
export { Spinner } from './spinner/spinner';
export { SymbolChoice } from './symbol-choice/symbol-choice';
export { TurnIndicator } from './turn-indicator/turn-indicator';
