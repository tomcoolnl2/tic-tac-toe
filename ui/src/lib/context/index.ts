export * from './content/content';
export * from './auth/auth';
