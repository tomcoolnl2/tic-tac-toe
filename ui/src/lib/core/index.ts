export * from './canvas';
export * from './grid';
export * from './flex-box';
export * from './divider';
