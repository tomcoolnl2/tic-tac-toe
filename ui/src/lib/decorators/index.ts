export { withTheme } from './withTheme';
export type { ContainerProps } from './withContainer';
export { withContainer } from './withContainer';
