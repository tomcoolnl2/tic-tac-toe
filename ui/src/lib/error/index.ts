export * from './error-boundary';
export { AuthError } from '../context/auth/error';
