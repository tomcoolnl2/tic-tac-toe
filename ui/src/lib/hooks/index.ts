export * from './useAnimationFrame';
export * from './useBehaviorSubjectState';
export * from './useScreenOrientation';
export * from './useSettingsHandlers';
export * from './useInterfaceHandlers';
export * from './useTimer';
