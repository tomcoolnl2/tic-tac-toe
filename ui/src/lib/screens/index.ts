export { LoadingScreen } from './loading';
export { LoginScreen } from './login';
export { ErrorScreen } from './error';
export { GameScreen } from './game';
export { SettingsScreen } from './settings';
export { RestartModalScreen } from './reload';
export { GameOverModalScreen } from './game-over';
