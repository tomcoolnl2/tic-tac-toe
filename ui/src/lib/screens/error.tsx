import React from 'react';
import { FlexBox, Divider } from '../core';
import { AppLogo, Button } from '../components';

export const ErrorScreen: React.FC = () => {
	return (
		<div className="screen-error" data-testid="error-boundary">
			<FlexBox direction="column" alignItems="center">
				<AppLogo />
				<Divider invisible margin="vertical-l" />
				<span>Something went wrong!</span>
				<Divider invisible margin="vertical-s" />
				<sub>Please, try again later...</sub>
				<Divider invisible margin="vertical-l" />
				<Button onClick={() => window.location.reload()}>
					Try again
				</Button>
			</FlexBox>
		</div>
	);
};
