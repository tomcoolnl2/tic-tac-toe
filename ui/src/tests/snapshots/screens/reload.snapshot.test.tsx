import { render } from '@testing-library/react';
import { ReloadModalScreen } from '../../../lib/screens';

describe('ReloadModalScreen screen component snapshot test', () => {
	const content = {
		cta1: 'No, Cancel',
		cta2: 'Yes, Restart',
		title: 'Restart game?',
	};
	it('should match the snapshot', () => {
		const fn = jest.fn();
		const { asFragment } = render(
			<ReloadModalScreen content={content} handleQuitGame={fn} closeModalScreen={fn} />
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
