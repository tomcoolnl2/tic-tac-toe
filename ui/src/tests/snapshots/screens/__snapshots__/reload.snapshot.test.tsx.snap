// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestartModalScreen screen component snapshot test should match the snapshot 1`] = `
<DocumentFragment>
  <h1>
    Restart game?
  </h1>
  <hr
    class="divider full-width horizontal margin-vertical-l"
  />
  <div
    class="container cols-2 col-gap-l"
    style="--row-height: 1fr; --col-width: 1fr;"
  >
    <button
      class="button button-light"
    >
      No, Cancel
    </button>
    <button
      class="button button-secondary"
    >
      Yes, Restart
    </button>
  </div>
</DocumentFragment>
`;
